build.assets/Dockerfile-ct-ng

```diff
@@ -0,0 +1,57 @@
+# Base on https://github.com/crosstool-ng/crosstool-ng/blob/8825cfc2abf696395bd27bd0e7cea3653004280b/testing/docker/ubuntu22.04/Dockerfile
+FROM ubuntu:22.04 as ct-ng
+
+ARG CTNG_UID=1000
+ARG CTNG_GID=1000
+
+RUN groupadd -g $CTNG_GID ctng
+RUN useradd -d /home/ctng -m -g $CTNG_GID -u $CTNG_UID -s /bin/bash ctng
+
+# Non-interactive configuration of tzdata
+ENV DEBIAN_FRONTEND noninteractive
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+RUN { echo 'tzdata tzdata/Areas select Etc'; echo 'tzdata tzdata/Zones/Etc select UTC'; } | debconf-set-selections
+
+RUN apt-get update
+RUN apt-get install -y gcc g++ gperf bison flex texinfo help2man make libncurses5-dev \
+    python3-dev autoconf automake libtool libtool-bin gawk wget bzip2 xz-utils unzip \
+    patch libstdc++6 rsync git meson ninja-build
+
+USER ctng
+WORKDIR /home/ctng
+
+RUN wget https://github.com/crosstool-ng/crosstool-ng/releases/download/crosstool-ng-1.26.0/crosstool-ng-1.26.0.tar.bz2 && \
```

Comment:
I'd recommend we verify the checksum of this tarball.  This isn't as important because we have a pinned version and can probably trust GitHub/Microsoft to not get hacked.  We're more protecting against the crosstools-ng org/user being compromised.

Reply:
https://github.com/gravitational/teleport/pull/36066#discussion_r1437267058